### PR TITLE
For #1002: Upgrade qulice to 0.17.4.

### PR DESCRIPTION
**Pull request by paulodamaso**
For #1002: Upgraded qulice to 0.17.4 version, corrected wrong tags and exclude values in `pom.xml`.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#1406](https://github.com/zerocracy/farm/issues/1406) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#1406](https://github.com/zerocracy/farm/issues/1406) is assigned to @krzyk[/z](https://www.0crat.com/u/krzyk), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/1912213814); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/zerocracy/tk/project/RqProject.java [hunk 1]
```diff
@@ -65,7 +65,8 @@ public RqProject(final Farm farm, final RqRegex req,
                 final String user = new RqUser(farm, req, false).value();
                 final Roles roles = new Roles(project).bootstrap();
                 final Roles admins = new Roles(pmo).bootstrap();
-                if (required.length > 0 && !roles.hasRole(user, required)
+                if (new LengthOf(new IterableOf<>(required)).intValue() > 0
```
Comment (krzyk) [inline src/main/java/com/zerocracy/tk/project/RqProject.java]: @paulodamaso why the changes here? Are they needed for qulice upgrade? If not I would revert them, there are already big changes in this PR.

File: src/main/java/com/zerocracy/tk/project/RqProject.java [hunk 2]
```diff
@@ -77,7 +78,7 @@ public RqProject(final Farm farm, final RqRegex req,
                         )
                     );
                 }
-                if (required.length == 0
+                if (new LengthOf(new IterableOf<>(required)).intValue() == 0
```
Comment (krzyk) [inline src/main/java/com/zerocracy/tk/project/RqProject.java]: @paulodamaso are this changes need for new qulice?

**Comment by paulodamaso** [comment]
@krzyk New quilice triggers the following:
```
[INFO] PMD: C:\git\zerocracy-farm\src\main\java\com\zerocracy\tk\project\RqProject.java[81-81]:
Use String.isEmpty() when checking for empty string (UseStringIsEmptyRule)
```
But we are not checking if a `String` is empty; it can be a `String` or an array of `String`. So, instead ignoring this check (I try to avoid using annotations to ignore checks whenever is possible) I chose to fix the problem.

**Comment by paulodamaso** [comment]
@krzyk New quilice triggers the following:
```
[INFO] PMD: C:\git\zerocracy-farm\src\main\java\com\zerocracy\tk\project\RqProject.java[81-81]:
Use String.isEmpty() when checking for empty string (UseStringIsEmptyRule)
```
But we are not checking if a `String` is empty; it can be a `String` or an array of `String`. So, instead ignoring this check (I try to avoid using annotations to ignore checks whenever is possible) I chose to fix the problem.

**Comment by krzyk** [comment]
@paulodamaso thanks

**Comment by krzyk** [comment]
@rultor merge

**Comment by g4s8** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @krzyk[/z](https://www.0crat.com/u/krzyk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#1406](https://github.com/zerocracy/farm/issues/1406) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Order was finished, quality is "good": +31 point(s) just awarded to @krzyk[/z](https://www.0crat.com/u/krzyk)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)
